tests/components/alexa/test_capabilities.py

```diff
@@ -243,6 +278,102 @@ async def test_api_select_input(
     assert call.data["source"] == source_list[idx]
 
 
+@pytest.mark.parametrize(
+    (
+        "target_activity",
+        "activity_list",
+        "current_activity_index",
+        "target_activity_index",
+    ),
+    [
+        ("TV", ["TV", "MUSIC", "DVD"], 1, 0),
+        ("MUSIC", ["TV", "MUSIC", "DVD", 1000], 0, 1),
+        ("DVD", ["TV", "MUSIC", "DVD", None], 0, 2),
+        ("BAD DEVICE", ["TV", "MUSIC", "DVD"], 0, None),
+        ("TV", ["TV"], 0, 0),
+        ("BAD DEVICE", [], None, None),
+    ],
+)
+async def test_api_select_activity(
+    hass: HomeAssistant,
+    target_activity: str,
+    activity_list: list[str],
+    current_activity_index: int | None,
+    target_activity_index: int | None,
+) -> None:
+    """Test api set activity process."""
+    curr_activty = (
+        activity_list[current_activity_index] if current_activity_index else "None"
```

Comment:
If `current_activity_index` is 0 we're defaulting `curr_activty` to `"None"` and not the using the first item in the activity list. Is that really intended?

Reply:
Instead of this condition I suggest we add another parameter for `curr_activity`.

Reply:
`"None"`, ie a stringified `None`, seems like a weird current activity. Why aren't we simply using `None`?

Reply:
The use of `None` here is an added void preset that Alexa needs as it does not accept a single preset.

Reply:
I was wrong about the preset. That is not what is happening here, removed the condition and used `None`.
https://github.com/home-assistant/core/pull/121627